subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedCustomTaskExecutionIntegrationTest.groovy

```diff
@@ -590,6 +590,131 @@ class CachedCustomTaskExecutionIntegrationTest extends AbstractIntegrationSpec i
         output.contains "Not caching task ':customTask' because no valid cache key was generated"
     }
 
+    def "task stays up-to-date after loaded from cache"() {
+        file("input.txt").text = "input"
+        buildFile << defineProducerTask()
+
+        withBuildCache().succeeds "producer"
+
+        when:
+        cleanBuildDir()
+        withBuildCache().succeeds "producer"
+        then:
+        skippedTasks as List == [":producer"]
+
+        when:
+        withBuildCache().succeeds "producer", "--info"
```

Comment:
Cacheability is determined even for up to date tasks, so we would still see the message.